io/disk/vfc-tests/virtual_fc.py

```diff
@@ -0,0 +1,531 @@
+                vfc_dic["paths"] = self.get_paths(vfc_dic["vfc_client"])
+                self.dic_list.append(vfc_dic)
+
+        self.log.info("complete list : %s" % self.dic_list)
+
+    def login(self, ipaddr, username, password):
+        '''
+        SSH Login method for remote server
+        '''
+        pxh = pxssh.pxssh()
+        # Work-around for old pxssh not having options= parameter
+        pxh.SSH_OPTS = pxh.SSH_OPTS + " -o 'StrictHostKeyChecking=no'"
+        pxh.SSH_OPTS = pxh.SSH_OPTS + " -o 'UserKnownHostsFile /dev/null' "
+        pxh.force_password = True
+
+        pxh.login(ipaddr, username, password)
+        pxh.sendline()
+        pxh.prompt(timeout=60)
+        # Ubuntu likes to be "helpful" and alias grep to
+        # include color, which isn't helpful at all. So let's
+        # go back to absolutely no messing around with the shell
+        pxh.set_unique_prompt()
+        pxh.prompt(timeout=60)
+        self.pxssh = pxh
+
+    def run_command(self, command, timeout=300):
+        '''
+        SSH Run command method for running commands on remote server
+        '''
+        self.log.info("Running the command on hmc %s", command)
+        con = self.pxssh
+        con.sendline(command)
+        con.expect("\n")  # from us
+        con.expect(con.PROMPT, timeout=timeout)
+        output = con.before.decode('utf-8').splitlines()
+        con.sendline("echo $?")
+        con.prompt(timeout)
+        return output
+
+    @staticmethod
+    def get_mcp_component(component):
+        '''
+        probes IBM.MCP class for mentioned component and returns it.
+        '''
+        for line in process.system_output('lsrsrc IBM.MCP %s' % component,
+                                          ignore_status=True, shell=True,
+                                          sudo=True).decode("utf-8") \
+                                                    .splitlines():
+            if component in line:
+                return line.split()[-1].strip('{}\"')
+        return ''
+
+    @staticmethod
+    def get_partition_name(component):
+        '''
+        get partition name from lparstat -i
+        '''
+
+        for line in process.system_output('lparstat -i', ignore_status=True,
+                                          shell=True,
+                                          sudo=True).decode("utf-8") \
+                                                    .splitlines():
+            if component in line:
+                return line.split(':')[-1].strip()
+        return ''
+
+    def test_unmap_map(self):
+        '''
+        Remove and add vfc interfaces Dynamically from HMC
+        '''
+        for _ in range(self.count):
+            for vfc_dic in self.dic_list:
+                self.vfchost_map_unmap("unmap", vfc_dic)
+                time.sleep(self.opp_sleep_time)
+                self.vfchost_map_unmap("map", vfc_dic)
+                # sleeping between the operation. can be enhanced in future.
```

Comment:
@Naresh-ibm We need the reason for sleep, and why it is the only possible solution now.